platforms_test.go

```diff
@@ -316,7 +346,7 @@ func TestParseSelector(t *testing.T) {
 				}
 			}
 
-			formatted := Format(p)
```

Comment:
Because the `Format` tests have been turned into `FormatV2` tests, we no longer have test coverage for Format, so I'd suggest, if you have the time, either adding a couple of quick new tests for Format (including covering that OSVersion is lost) or expand this test suite with an expected `formattedv1` and expected v1roundtrip for those cases where it's different, and verify it's the same when it's expected to be.